build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/EmbeddedProviderExtension.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal;
+
+import org.elasticsearch.gradle.transform.UnzipTransform;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.attributes.Attribute;
+import org.gradle.api.file.Directory;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.Sync;
+
+import static org.elasticsearch.gradle.internal.conventions.GUtils.capitalize;
+import static org.elasticsearch.gradle.util.GradleUtils.getJavaSourceSets;
+import static org.gradle.api.artifacts.type.ArtifactTypeDefinition.ARTIFACT_TYPE_ATTRIBUTE;
+import static org.gradle.api.artifacts.type.ArtifactTypeDefinition.DIRECTORY_TYPE;
+import static org.gradle.api.artifacts.type.ArtifactTypeDefinition.JAR_TYPE;
+
+public class EmbeddedProviderExtension {
+
+    private static final Attribute<Boolean> IMPL_ATTR = Attribute.of("is.impl", Boolean.class);
+
+    private final Project project;
+
+    public EmbeddedProviderExtension(Project project) {
+        this.project = project;
+    }
+
+    void impl(String implName, Project implProject) {
+        String projectName = implProject.getName();
+        String capitalName = capitalize(projectName);
+
+        String configName = "embedded" + capitalName;
+        Configuration implConfig = project.getConfigurations().create(configName);
```

Comment:
I think we can just create a detached configuration here since it's not intended to be accessed in any other way.